docs/index.md

```diff
@@ -27,8 +27,9 @@ provider "auth0" {
 ## Argument Reference
 
 * `domain` - (Required) Your Auth0 domain name. It can also be sourced from the `AUTH0_DOMAIN` environment variable.
-* `client_id` - (Required) Your Auth0 client ID. It can also be sourced from the `AUTH0_CLIENT_ID` environment variable.
-* `client_secret` - (Required) Your Auth0 client secret. It can also be sourced from the `AUTH0_CLIENT_SECRET` environment variable.
+* `client_id` - (Optional) Your Auth0 client ID. It can also be sourced from the `AUTH0_CLIENT_ID` environment variable.
+* `client_secret` - (Optional) Your Auth0 client secret. It can also be sourced from the `AUTH0_CLIENT_SECRET` environment variable.
+* `auth0_token` - (Optional) Your Auth0 [management api access token]. It can also be sourced from the `AUTH0_TOKEN` environment variable. Can be used instead of `client_id` + `client_secret`. If both are specified, `management_token` will be used over `client_id` + `client_secret` fields.
```

Comment:
Thanks for updating the documentation here. While these arguments are still technically optional, I think we can make it more clear that there are two methods of authentication that could occur. I think this could end up being a point of confusion for new devs.

Reply:
Hey, will try to take another stab at it. I was taking inspiration from how other terraform providers handle this:
+ [cloudflare](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs)
    ![image](https://user-images.githubusercontent.com/17842707/152892684-7d72edcf-c87b-4f69-af11-6adc93f0debb.png)

+ [gce](https://registry.terraform.io/providers/hashicorp/google/3.29.0/docs/guides/provider_reference)

    ![image](https://user-images.githubusercontent.com/17842707/152892788-71d427bd-53fd-4a03-acb6-99ee9fa976e9.png)

so this seems at least 'idomatic', but agreed with your statement

Reply:
Completely understood. I was mostly thinking out loud. Let's focus on the technical aspects in this PR and if I think of changes to the documentation, I'll apply those.